src/Orleans/Serialization/RemoteNonDeserializableException.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Runtime.Serialization;
+using Orleans.Runtime;
+
+namespace Orleans.Serialization
+{
+    using System;
+    using System.Text;
+
+    /// <summary>
+    /// Represents an exception which cannot be fully deserialized.
+    /// </summary>
+    [Serializable]
+    public class RemoteNonDeserializableException : OrleansException
+    {
+        public RemoteNonDeserializableException() { }
+
+        /// <summary>
+        /// Gets the type name of the original <see cref="Exception"/> represented by this instance.
+        /// </summary>
+        public string OriginalTypeName { get; internal set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the original type of this exception can be reconstructed during
+        /// deserialization.
+        /// </summary>
+        internal bool CanAttemptReconstruction { get; set; }
+
+        /// <summary>
+        /// Gets or sets the additional data deserialized alongside this 
+        /// </summary>
+        internal byte[] AdditionalData { get; set; }
```

Comment:
Quite suspicious of the value of this.  It's internal, so applications can't even use it.  Data is packed in binary which application will likely not know what to do with.  What if one of the property types is not serializable?  A property bag may have some value, as at least that could contain named properties that the application layer can inspect.  I'd drop for now as type, message, stack should be suffice, but just my opinion.